osu.Game/Overlays/Toolbar/ToolbarClock.cs

```diff
@@ -3,56 +3,84 @@
 
 using osu.Framework.Allocation;
 using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
 using osu.Framework.Input.Events;
 using osu.Game.Configuration;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
 using osuTK;
+using osuTK.Graphics;
 
 namespace osu.Game.Overlays.Toolbar
 {
-    public class ToolbarClock : CompositeDrawable
+    public class ToolbarClock : OsuClickableContainer
     {
         private Bindable<ToolbarClockDisplayMode> clockDisplayMode;
 
+        protected Box HoverBackground;
+        private readonly Box flashBackground;
+        private readonly FillFlowContainer clockContainer;
+
         private DigitalClockDisplay digital;
         private AnalogClockDisplay analog;
 
         public ToolbarClock()
+            : base(HoverSampleSet.Toolbar)
         {
             RelativeSizeAxes = Axes.Y;
             AutoSizeAxes = Axes.X;
 
-            Padding = new MarginPadding(10);
-        }
-
-        [BackgroundDependencyLoader]
-        private void load(OsuConfigManager config)
-        {
-            clockDisplayMode = config.GetBindable<ToolbarClockDisplayMode>(OsuSetting.ToolbarClockDisplayMode);
-
-            InternalChild = new FillFlowContainer
+            Children = new Drawable[]
             {
-                RelativeSizeAxes = Axes.Y,
-                AutoSizeAxes = Axes.X,
-                Direction = FillDirection.Horizontal,
-                Spacing = new Vector2(5),
-                Children = new Drawable[]
+                HoverBackground = new Box
                 {
-                    analog = new AnalogClockDisplay
-                    {
-                        Anchor = Anchor.CentreLeft,
-                        Origin = Anchor.CentreLeft,
-                    },
-                    digital = new DigitalClockDisplay
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = OsuColour.Gray(80).Opacity(180),
+                    Blending = BlendingParameters.Additive,
+                    Alpha = 0,
+                },
+                flashBackground = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Alpha = 0,
+                    Colour = Color4.White.Opacity(100),
+                    Blending = BlendingParameters.Additive,
+                },
+
+                clockContainer = new FillFlowContainer
```

Comment:
Why was all of this moved from `load()`? Please put it back there if there is no good reason for the move.

Reply:
* That's not a "crash", it's a compilation failure.
* As noted in the other review comment you don't actually need `clockContainer` for anything. But if you did, the way to proceed would be to... just not make it `readonly`.

Reply:
> That's not a "crash", it's a compilation failure.
Sorry, I'm not used to differentiating those. When I get some kind if error I'm used to calling it a crash.
> As noted in the other review comment you don't actually need clockContainer for anything. But if you did, the way to proceed would be to... just not make it readonly.
Makes sense, I just assumed there was some kind of reason because it was done differently elsewhere.

Reply:
@Sublimelime please, there is no need to pile on further. i believe the point has been made already.